modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/entity/v1_0/ObjectEntryEntityModel.java

```diff
@@ -174,6 +166,48 @@ private ObjectDefinition _getRelatedObjectDefinition(
 			objectRelationship.getObjectDefinitionId2());
 	}
 
+	private List<EntityField> _getRelatedObjectDefinitionEntityFields(
+			ObjectRelationship objectRelationship,
+			ObjectDefinition objectDefinition)
+		throws Exception {
+
+		_relatedObjectRelationships.add(
+			objectRelationship.getObjectRelationshipId());
+
+		ObjectDefinition relatedObjectDefinition = _getRelatedObjectDefinition(
+			objectDefinition, objectRelationship);
+
+		Map<String, EntityField> relatedObjectDefinitionEntityFieldMap =
+			_getStringEntityFieldMap(
+				ObjectFieldLocalServiceUtil.getObjectFields(
+					relatedObjectDefinition.getObjectDefinitionId()));
+
+		ArrayList<EntityField> relatedObjectDefinitionEntityFields =
+			new ArrayList<>(relatedObjectDefinitionEntityFieldMap.values());
+
+		List<ObjectRelationship> relatedObjectDefinitionObjectRelationships =
+			ObjectRelationshipLocalServiceUtil.getAllObjectRelationships(
```

Comment:
Is this working only for depth 2 or for every kind of depth? I have the feeling that it is just depth 2, but to be honest, I don't know if that is the requirement :thinking: